File: W3Schools/Backend/C/13_WhileLoop/13_WhileLoop.c

// C: W3Schools - Main Tutorial
// Section 13: While Loop

#include <stdio.h>

int main() {

    int i = 0;

    // The while loop loops through a block of code as long as a specified condition is true:
    while (i < 5) {
        printf("%d\n", i);
        i++;
    }

    i = 0;

    // Similarly, a do-while loop executes the code block once before evaluating the condition:
    do {
        printf("%d\n", i);
        i++;
    } while (i < 5);

    return 0;
}

File: W3Schools/Backend/R/13_WhileLoop.R

# R: W3Schools - Main Tutorial
# Section 13: While Loop

i <- 0

# While loops run until the condition is false:
while (TRUE) {
  i <- i + 1
  if (i == 3) { # The next keyword skips the current iteration.
    next
  }

  if (i > 5) {
    break # The break keyword stops the loop.
  }
  print(i)
} 